jerry-main/main-utils.c

```diff
@@ -44,6 +44,183 @@ main_register_global_function (const char *name_p, /**< name of the function */
+
+/**
+ * $262.createRealm
+ *
+ * A function which creates a new ECMAScript Realm, defines this API on the new realm's global object,
+ * and returns the $262 property of the new realm's global object
+ *
+ * @return the $262 property of the new realm's global object
+ */
+static jerry_value_t
+test262_create_realm (const jerry_value_t func_obj_val, /**< function object */
+                      const jerry_value_t this_p, /**< this arg */
+                      const jerry_value_t args_p[], /**< function arguments */
+                      const jerry_length_t args_cnt) /**< number of function arguments */
+{
+  (void) func_obj_val; /* unused */
+  (void) this_p; /* unused */
+  (void) args_p; /* unused */
+  (void) args_cnt; /* unused */
+
+  /* TODO: support different realms */
+  jerry_value_t global_obj = jerry_get_global_object ();
+  jerry_value_t prop_name = jerry_create_string ((const jerry_char_t *) "$262");
+  jerry_value_t result = jerry_get_property (global_obj, prop_name);
+
+  jerry_release_value (prop_name);
+  jerry_release_value (global_obj);
+
+  return result;
+} /* test262_create_realm */
+
+/**
+ * $262.detachArrayBuffer
+ *
+ * A function which implements the DetachArrayBuffer abstract operation
+ *
+ * @return null value - if success
+ *         value marked with error flag - otherwise
+ */
+static jerry_value_t
+test262_detach_array_buffer (const jerry_value_t func_obj_val, /**< function object */
+                             const jerry_value_t this_p, /**< this arg */
+                             const jerry_value_t args_p[], /**< function arguments */
+                             const jerry_length_t args_cnt) /**< number of function arguments */
+{
+  (void) func_obj_val; /* unused */
+  (void) this_p; /* unused */
+
+  if (args_cnt < 1 || !jerry_value_is_arraybuffer (args_p[0]))
+  {
+    return jerry_create_error (JERRY_ERROR_TYPE, (jerry_char_t *) "Expected an ArrayBuffer object");
+  }
+
+  /* TODO: support the optional 'key' argument */
+
+  return jerry_detach_arraybuffer (args_p[0]);
+} /* test262_detach_array_buffer */
+
+/**
+ * $262.evalScript
+ *
+ * A function which accepts a string value as its first argument and executes it
+ *
+ * @return completion of the script parsing and execution.
+ */
+static jerry_value_t
+test262_eval_script (const jerry_value_t func_obj_val, /**< function object */
+                     const jerry_value_t this_p, /**< this arg */
+                     const jerry_value_t args_p[], /**< function arguments */
+                     const jerry_length_t args_cnt) /**< number of function arguments */
+{
+  (void) func_obj_val; /* unused */
+  (void) this_p; /* unused */
+
+  if (args_cnt < 1 || !jerry_value_is_string (args_p[0]))
+  {
+    return jerry_create_error (JERRY_ERROR_TYPE, (jerry_char_t *) "Expected a string");
+  }
+
+  jerry_size_t str_size = jerry_get_utf8_string_size (args_p[0]);
+  jerry_char_t *str_buf_p = jerry_heap_alloc (str_size * sizeof (jerry_char_t));
```

Comment:
I'm not sure it is a good idea to use heap alloc here. What are we using when loading normal scripts form files?